project/CassandraSparkBuild.scala

```diff
@@ -41,20 +50,25 @@ object CassandraSparkBuild extends Build {
     ).copy(dependencies = Seq(embedded % "test->test;it->it,test;")
   ) configs (IntegrationTest, ClusterIntegrationTest)
 
-  lazy val jconnector = Project(
-    id = s"$namespace-java",
-    base = file(s"$namespace-java"),
-    settings = connector.settings,
-    dependencies = Seq(connector % "compile;runtime->runtime;test->test;it->it,test;provided->provided")
+  lazy val jconnector = CrossScalaVersionsProject(
+    name = s"$namespace-java",
+    conf = assembledSettings ++ Seq(libraryDependencies ++= Dependencies.connector ++ Seq(
+        "org.scala-lang" % "scala-reflect"  % scalaVersion.value,
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % "test,it"), compileOrder in Compile := CompileOrder.ScalaThenJava)
+    ).copy(dependencies = Seq(connector % "compile;runtime->runtime;test->test;it->it,test;provided->provided")
```

Comment:
that's not just line too long, but this whole expression is hard to read; some parts of it should be taken out to another variable